### PR TITLE
Autocomplete member in groupe input

### PR DESCRIPTION
**Pull request by LucieLDa**
UPDATE Group component
When adding an user to group, suggestion and autocompletion is enabled

### CODE REVIEW

**Review by Fatimata-dev — APPROVED** [review]

File: src/app/components/edit-group-dialog/edit-group-dialog.component.ts [hunk 1]
```diff
@@ -1,23 +1,41 @@
...
+            MatFormFieldModule, 
+            MatIconModule, 
+            CommonModule, 
+            MatAutocompleteModule,
+            AsyncPipe
+          ],
   templateUrl: './edit-group-dialog.component.html',
   styleUrl: './edit-group-dialog.component.scss'
 })
 export class EditGroupDialogComponent implements OnInit{
```
Comment (Benja-san) [inline src/app/components/edit-group-dialog/edit-group-dialog.component.ts]: you don't need to implement onInit anymore (GPT generated code ?)

File: src/app/components/edit-group-dialog/edit-group-dialog.component.ts [hunk 2]
```diff
@@ -1,23 +1,41 @@
...
   groupForm: FormGroup;
   totalMembers: number = 0;
+  currentMember : number = 0;
+
+  private userService : UserService = inject(UserService);
+  users : User[] = [];
+  filteredUsers !: Observable<User[]>;
```
Comment (Benja-san) [inline src/app/components/edit-group-dialog/edit-group-dialog.component.ts]: use $ at the end of a variable if you are using observable

File: src/app/components/edit-group-dialog/edit-group-dialog.component.ts [hunk 3]
```diff
@@ -57,8 +78,8 @@ export class EditGroupDialogComponent implements OnInit{
 
   addMember(): void {
     this.membersFormArray.push(this.fb.group({
```
Comment (Benja-san) [inline src/app/components/edit-group-dialog/edit-group-dialog.component.ts:80]: refactor this duplicated code into a method reusable in add and load

**Review by Benja-san — CHANGES_REQUESTED** [review]

File: src/app/components/edit-group-dialog/edit-group-dialog.component.ts (hunk 1 above)
Comment (LucieLDa) [reply src/app/components/edit-group-dialog/edit-group-dialog.component.ts]: That's not my code, I am not the one who created EditGroupDialogComponent. I did not use GPT once during this project. All I did was add a new functionality in an older component previously reviewed.

File: src/app/components/edit-group-dialog/edit-group-dialog.component.ts (hunk 3 above)
Comment (LucieLDa) [reply src/app/components/edit-group-dialog/edit-group-dialog.component.ts:80]: Too different to refactor

**Review by Benja-san — APPROVED** [review]

**Pull request merged**